includes/Embed_Base.php

```diff
@@ -34,96 +34,30 @@
 /**
  * Embed block class.
  */
-class Embed_Block {
+class Embed_Base {
```

Comment:
We're getting there!

Now `Embed_Base` does three things:

- Register some essential scripts upon `init` that are used on the frontend (block & shortcode), and on the backend (block)
- Filter KSES to allow `amp-story-player` elements
- Serve as a base class for the blocks and shortcodes

I think it should only do the latter, serve as a base class. The other part should be in a separate class.

Otherwise if any of the block/shortcode classes inadvertently  call `parent::init()`, it will try to register the scripts & filters twice.

WDYT?

Reply:
Why would you call `parent::init()`?  We could move some of the logic out into another class, but it means spreading the code out. Not sure I like this, as it may make it harder to find later. 

In my head, I was thinking, what needs to shared for any interface for this. Say for example, we implement a widget that expends this functionality. 

- The render method needs to be used by every implementation, which is why it is still in base. 
- `STORY_PLAYER_HANDLE` is likely needed by most implementations.

Reply:
> Why would you call `parent::init()`?

As I said, it could happen _inadvertently_. Maybe in the future someone else contributes a new block/shortode and _inadvertently_ adds `parent::init()`, as one usually often does in OOP, and it's not caught in code review.

----

> The `render` method needs to be used by every implementation, which is why it is still in base.

Yes.

> `STORY_PLAYER_HANDLE` is likely needed by most implementations.

Yes, but this script only needs to be registered **_once_**, globally. In a separate location.

Reply:
>As I said, it could happen inadvertently. Maybe in the future someone else contributes a new block/shortode and inadvertently adds parent::init(), as one usually often does in OOP, and it's not caught in code review. 

This seems like very edge case to me. I dont think we normally write code, with "How can someone in the future how doesnt know what they are doing breaks our code" in mind. Honestly you could drive yourself insane with that line of thinking. 

Why not move the registering of style/script it's own method? Nothing wrong happens even if it is called twice anyway...

Reply:
Fair enough, but I still think asset registration should be in something like an `Assets` class. Not in an Embed_Base class that serves as a base class for a bunch of others _and_ is also accessible via `get_plugin_instance()->embed_base`

Reply:
Okay, will these scripts be used outside of this class? If not, then should remain in there, otherwise, they should be moving. If there are not used at the moment, can this change be made in at future date? As and when it is required.

Reply:
Most likely, yes.

We can do it in a follow-up PR if that's easier.

Reply:
I think the best course, is add a todo to code and follow up if we need to move to another class and only then.